overlord/snapstate/component.go

```diff
@@ -78,33 +78,46 @@ func InstallComponentPath(st *state.State, csi *snap.ComponentSideInfo, info *sn
 		},
 	}
 
-	return doInstallComponent(st, &snapst, compSetup, snapsup, "")
+	componentTS, err := doInstallComponent(st, &snapst, compSetup, snapsup, "")
+	if err != nil {
+		return nil, err
+	}
+
+	return componentTS.ts, nil
 }
 
 type componentInstallFlags struct {
 	RemoveComponentPath bool `json:"remove-component-path,omitempty"`
 	SkipProfiles        bool `json:"skip-profiles,omitempty"`
 }
 
+type componentInstallTaskSet struct {
+	compSetupTask      *state.Task
+	beforeLink         []*state.Task
+	linkToHook         []*state.Task
+	postOpHookAndAfter []*state.Task
+	ts                 *state.TaskSet
+}
+
 // doInstallComponent might be called with the owner snap installed or not.
-func doInstallComponent(st *state.State, snapst *SnapState, compSetup *ComponentSetup,
-	snapsup *SnapSetup, fromChange string) (*state.TaskSet, error) {
+func doInstallComponent(st *state.State, snapst *SnapState, compSetup ComponentSetup,
+	snapsup SnapSetup, fromChange string) (componentInstallTaskSet, error) {
```

Comment:
Are there unit tests that check that the tasks are generated for components that do have the hooks?

Reply:
Yes, since the tasks are always created regardless if the component has hooks or not, all of the tests have to consider the hook tasks.